### PR TITLE
OperationalInsights: 2022-10-01 stable version.

### PR DESCRIPTION
**Pull request by dulikvor**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [x] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/operationalinsights/resource-manager/readme.go.md [hunk 1]
```diff
@@ -27,6 +27,7 @@ batch:
   - tag: package-2021-12-01-preview
   - tag: package-2020-10
   - tag: package-2020-08
+  - tag: package-2022-10
```
Comment (ArcturusZhang) [inline specification/operationalinsights/resource-manager/readme.go.md:30]: could we revert the changes in this file? These changes serve the track 1 Go SDK which is going to be deprecated very soon and we no longer release new versions for it.

Comment (dulikvor) [reply specification/operationalinsights/resource-manager/readme.go.md:30]: Done.

**Comment by dulikvor** [comment]
Avocado failures are false Alarm. no operation was deprecated.
confusion originates from casing change "resourcegroups" -> "resourceGroups".

**Comment by zizw123** [comment]
Is anything being copied from previous API version? Or is everything new in this API version.

If anything is copied over, the first commit needs to be the previous api version and the new changes should only be added in the subsequent commits. This is explained in detail here (If using Open API which is recommended) https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version or here(For manual process) : https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/83/Manual-Process-Adding-new-API-version .

**Comment by dulikvor** [comment]
> Is anything being copied from previous API version? Or is everything new in this API version.
> 
> If anything is copied over, the first commit needs to be the previous api version and the new changes should only be added in the subsequent commits. This is explained in detail here (If using Open API which is recommended) https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version or here(For manual process) : https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/83/Manual-Process-Adding-new-API-version .

Most were copied, there are a few changes, you can see the diff quite easily by reviewing the commits.

**Comment by dulikvor** [comment]
Just to clarify the commits:
First commit -> A copy of the previous stable version commit + api version change to 2022-10-01.
Second commit and above -> actual changes and diff which should be reviewied.

**Comment by raosuhas** [comment]
Thansk @dulikvor  , just FYI for future PRs please avoid doing the "+ api version change to 2022-10-01" so we are aware of the baseline commit directly.  The first commit should be previous api version with no changes. I will review this PR assuming that the api version is the only change made in the first commit

---
In reply to: [1230610433](https://github.com/Azure/azure-rest-api-specs/pull/20236#issuecomment-1230610433) [](http://example.com/codeflow?ancestors=1230610433,1230610433)

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Tables.json [hunk 2]
```diff
@@ -0,0 +1,847 @@
...
+                "description": "Table state is stable and without changes, table is unlocked and open for new updates."
+              }
+            ]
+          }
+        },
+        "retentionInDaysAsDefault": {
+          "readOnly": true,
+          "type": "boolean",
+          "description": "True - Value originates from workspace retention in days, False - Customer specific."
+        },
```
Comment (raosuhas) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Tables.json:794]: Consider using enums as explained here :

[https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review)

Replace boolean/switch properties with better enum
A Boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Even if you still believe [True, False] are the correct values for a property, you should use a string enum with values [True, False] instead of boolean. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [True, False, Unknown].

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Tables.json [hunk 3]
```diff
@@ -0,0 +1,847 @@
...
+        "searchResults": {
+          "$ref": "#/definitions/SearchResults",
+          "description": "Parameters of the search job that initiated this table."
+        },
+        "restoredLogs": {
+          "$ref": "#/definitions/RestoredLogs",
+          "description": "Parameters of the restore operation that initiated this table."
+        },
+        "resultStatistics": {
+          "readOnly": true,
```
Comment (raosuhas) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Tables.json:726]: Updating a field to readonly is a breaking change. Is this already signed off ?

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Workspaces.json [hunk 4]
```diff
@@ -0,0 +1,786 @@
...
+        "disableLocalAuth": {
+          "type": "boolean",
+          "x-nullable": true,
+          "description": "Disable Non-AAD based Auth."
+        }
+      },
+      "additionalProperties": true,
+      "description": "Workspace features."
+    },
+    "Identity": {
```
Comment (raosuhas) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Workspaces.json:623]: >Identity [](http://example.com/codeflow?start=5&length=8)

Cosider using common types for identity : 

[https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/managedidentity.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/managedidentity.json)

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Tables.json (hunk 3 above)
Comment (dulikvor) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Tables.json:726]: Change was made from public preview version and was never stable. breaking change is allowed in this case.

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Tables.json (hunk 2 above)
Comment (dulikvor) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Tables.json:794]: Server side is serializing a Boolean, can we be sure that client side which generates his client from the swagger will read the response correctly? Boolean in json will just return -> true instead of "True".

I think we should depict the server-side contract as it is.

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Workspaces.json (hunk 4 above)
Comment (dulikvor) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Workspaces.json:623]: Done.

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Tables.json (hunk 2 above)
Comment (raosuhas) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Tables.json:794]: As long as you set modelasstring property to true the client will have no trouble with this. modelling values as enums is pretty standard practise across azure services and is a pretty strong recommendation.

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Workspaces.json (hunk 4 above)
Comment (raosuhas) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Workspaces.json:623]: You can just use the "ManagedServiceIdentity" type

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Tables.json (hunk 2 above)
Comment (raosuhas) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Tables.json:794]: This is not a blocking comment. but strongly recommended just as an FYI

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Workspaces.json (hunk 4 above)
Comment (dulikvor) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Workspaces.json:623]: Done.

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Tables.json (hunk 2 above)
Comment (dulikvor) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Tables.json:813]: Made the change.

**Comment by akning-ms** [comment]
@jianyexi to help take a look why linter diff has exception.

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Workspaces.json [hunk 5]
```diff
@@ -0,0 +1,735 @@
...
+      "$ref": "../../../../../common-types/resource-management/v3/managedidentity.json",
```
Comment (jianyexi) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2022-10-01/Workspaces.json:624]: this is not valid usage, please remove this definition, and refer to ""../../../../../common-types/resource-management/v3/managedidentity.json" in the properties directly

**Comment by jianyexi** [comment]
> @jianyexi to help take a look why linter diff has exception.

There is schema validator error, already commented on how to fix it

**Comment by akning-ms** [comment]
@tadelesh @qingquanzhao @msyyc to confirm GO, JS, Python SDK breaking change for this PR

**Comment by msyyc** [comment]
approved for python breaking

**Review by ruowan — APPROVED** [review]
